tests/test_endpoints.py

```diff
@@ -1,24 +1,84 @@
 from requests_flask_adapter import Session
-from kirby.web import app_maker
 from pytest import fixture
+from datetime import datetime
+
+from kirby.web import app_maker
+from kirby.models import db, Environment, Job, JobType, Context, Script, Topic
 
 BASE_API = "http://some-test-server.somewhere"
 
 
+@fixture(scope="function")
+def webapp():
+    config = {
+        "TESTING": True,
+        "SQLALCHEMY_DATABASE_URI": "sqlite://",
+        "SQLALCHEMY_TRACK_MODIFICATIONS": False,
+    }
+    app = app_maker(config=config)
+    with app.app_context():
+        db.create_all()
+        yield app
+
+
 @fixture
-def session():
-    app = app_maker()
-    Session.register(BASE_API, app)
+def session(webapp):
+    Session.register(BASE_API, webapp)
     return Session()
 
 
-def test_it_register_a_job(session):
-    result = session.post(
+@fixture
+def db_populated():
```

Comment:
This is a bit too generic of a name. It does not help to know what kind of data you are putting in the database, so when building tests based on this fixture, one has to read the whole code to know if it's going to help or not in testing a feature.

Please find a more descriptive name (if you struggle finding a name, it might be because you're trying to do too many things at once as well).

Reply:
The tests have been rewritten, they now use factory to allow more flexibility for the future tests